### PR TITLE
Adding documentation for implementing bwc tests in plugins

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha <vachshah@amazon.com>

### Description
Adding documentation for backwards compatibility tests for plugins.
 
### Issues Resolved
[opensearch-build#223](https://github.com/opensearch-project/opensearch-build/issues/223)
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -66,4 +69,32 @@ https://app.codecov.io/gh/opensearch-project/REPO-NAME-HERE/settings/badge
 1. Add a `codecov.yml` file into the repository ([example](https://github.com/opensearch-project/k-NN/commit/f7d1985230ce851cb97a7e41d8bce32127a4f33b))
 2. Set the target code coverage properly in case it blocks your PR in Github checks.
 
-See Codecov docs to learn more about [codecov yaml file](https://docs.codecov.com/docs/codecov-yaml) and [common configurations](https://docs.codecov.com/docs/common-recipe-list).
\ No newline at end of file
+See Codecov docs to learn more about [codecov yaml file](https://docs.codecov.com/docs/codecov-yaml) and [common configurations](https://docs.codecov.com/docs/common-recipe-list).
+
+### Backwards Compatibility Testing
+
+The purpose of backwards compatibiity tests in plugins is to ensure that the plugin is safely upgraded when an OpenSearch node undergoes an upgrade. Backwards compatibility tests can be implemented in plugins by extending the OpenSearch bwc framework. The tests and the framework have been added in [`anomaly-detection`](https://github.com/opensearch-project/anomaly-detection) plugin which can be used as an example to add tests to other plugins.
```
Comment (dblock) [inline TESTING.md:76]: Nit: don't quote anomaly-detection and spell OpenSearch case-sensitively.

File: TESTING.md [hunk 2]
```diff
@@ -66,4 +69,32 @@ https://app.codecov.io/gh/opensearch-project/REPO-NAME-HERE/settings/badge
...
+
+The changes required in the OpenSearch bwc framework to support plugins bwc tests have been added as part of [opensearch#1051](https://github.com/opensearch-project/OpenSearch/pull/1051)
```
Comment (dblock) [inline TESTING.md:78]: I would say `See [OpenSearch#1051](https://github.com/opensearch-project/OpenSearch/pull/1051) for more information.`

File: TESTING.md [hunk 3]
```diff
@@ -66,4 +69,32 @@ https://app.codecov.io/gh/opensearch-project/REPO-NAME-HERE/settings/badge
...
+
+#### Types of BWC tests
+
+1. Mixed cluster test: This tests a cluster running with three nodes of a bwc OpenSearch and plugin version and upgrades a single node to the current version of OpenSearch and the plugin while the other two nodes are still on the old version. This creates a mixed cluster. The test checks that the correct version of the plugin is installed in the upgraded node and the plugin functionalities work as expected after the upgrade.
```
Comment (dblock) [inline TESTING.md:82]: old -> previous?

**Review by dblock — DISMISSED** [review]
Looks great! Just nits, don't worry about fixing if you don't think it's worth it.

**Comment by VachaShah** [comment]
> Looks great! Just nits, don't worry about fixing if you don't think it's worth it.

Thank you @dblock, addressed the comments :)

**Review by saratvemulapalli — APPROVED** [review]
This this awesome!!!
Thanks @VachaShah !

**Review by dblock — APPROVED** [review]

**Pull request merged**